React/Modules/RCTTiming.m

```diff
@@ -233,16 +233,18 @@ - (void)scheduleSleepTimer:(NSDate *)sleepTarget
     _sleepTimer = [[NSTimer alloc] initWithFireDate:sleepTarget
                                            interval:0
                                              target:[_RCTTimingProxy proxyWithTarget:self]
-                                           selector:@selector(timerDidFire)
+                                           selector:@selector(sleepTimerDidFire)
                                            userInfo:nil
                                             repeats:NO];
-    [[NSRunLoop currentRunLoop] addTimer:_sleepTimer forMode:NSDefaultRunLoopMode];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [[NSRunLoop currentRunLoop] addTimer:_sleepTimer forMode:NSDefaultRunLoopMode];
+    });
   } else {
     _sleepTimer.fireDate = [_sleepTimer.fireDate earlierDate:sleepTarget];
   }
 }
 
-- (void)timerDidFire
+- (void)sleepTimerDidFire
```

Comment:
This method has certain expectations on which thread it's called from. At least with the JSC executor it should definitely not be called from the main thread.